config/sql/migration/migrate.14.sql

```diff
@@ -0,0 +1,19 @@
+-- Don't modify this! Create a new migration instead--see docs/schema-migrations.adoc
+BEGIN TRANSACTION;
+
+CREATE TABLE target_images_data(image_data BLOB NOT NULL);
+CREATE TABLE target_images_info(filename TEXT UNIQUE, real_size INTEGER NOT NULL DEFAULT 0, sha256 TEXT NOT NULL DEFAULT "", sha512 TEXT NOT NULL DEFAULT "", image_id integer NOT NULL);
+
+INSERT INTO target_images_info (filename, real_size, sha256, sha512, image_id)
+SELECT filename, real_size, sha256, sha512, rowid FROM target_images;
+
+INSERT INTO target_images_data(rowid, image_data)
+SELECT rowid, image_data FROM target_images;
```

Comment:
From https://www.sqlite.org/rowidtable.html:

> If the rowid is not aliased by INTEGER PRIMARY KEY then it is not persistent and might change. In particular the VACUUM command will change rowids for tables that do not declare an INTEGER PRIMARY KEY. Therefore, applications should not normally access the rowid directly, but instead use an INTEGER PRIMARY KEY.

Looks like we are playing a dangerous game here. It looks like it would be safer to define and explicit `INTEGER PRIMARY KEY` or just reuse the filename as `PRIMARY KEY` in both tables

Reply:
Yes agree.